cub/device/dispatch/dispatch_reduce.cuh

```diff
@@ -51,6 +51,58 @@
 
 CUB_NAMESPACE_BEGIN
 
+namespace detail
+{
+namespace reduce
+{
+
+/**
+ * All cub::DeviceReduce::* algorithms are using the same implementation. Some of them, however,
+ * should use initial value only for empty problems. This struct is used to distinguish between
+ * mentioned cases.
+ */
```

Comment:
```suggestion
/**
 * All cub::DeviceReduce::* algorithms are using the same implementation. Some of them, however,
 * should use initial value only for empty problems. If this struct is used as initial value with
 * one of the `DeviceReduce` algorithms, the `init` value wrapped by this struct will only be used
 * for empty problems; it will not be incorporated into the aggregate of non-empty problems.
 */
```